frontend/src/metabase/dashboard/hoc/WithVizSettingsData.js

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable react/prop-types */
+import React from "react";
+import { connect } from "react-redux";
+import { withRouter } from "react-router";
+import _ from "underscore";
+
+import { getUserAttributes } from "metabase/selectors/user";
+import { getLinkTargets } from "metabase/lib/click-behavior";
+
+// This HOC give access to data referenced in viz settings.
+const WithVizSettingsData = ComposedComponent => {
```

Comment:
Moved from `visualizations` as it's only used in dashboards. The "composed" part of this component hasn't changed, but I've moved data loading for click behaviors to redux